src/current/v24.3/troubleshoot-lock-contention.md

```diff
@@ -0,0 +1,403 @@
+- How to [identify waiting and blocking transactions](#identify-waiting-and-blocking-transactions) by using the Insights page of the DB Console.
+- Possible ways to [remediate lock contention](#remediate-lock-contention).
+
+## Before you begin
+
+[Download]({% link releases/index.md %}) and [Install CockroachDB]({% link {{ page.version.version }}/install-cockroachdb-mac.md %}).
+
+### Terminal 1
+
+The examples in this tutorial will use three terminals, one for each transaction.
+
+In the first terminal, use the [`cockroach demo`]({% link {{ page.version.version }}/cockroach-demo.md %}) command to start a temporary, in-memory CockroachDB cluster of one node.
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach demo --no-example-database
+~~~
+
+It will open an interactive SQL shell to the cluster set to an empty database called `defaultdb`. This database is used for testing and some internal databases.
+
+To distinguish each terminal set the `application_name`. In the SQL shell in the first terminal, execute:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 1';  -- to distinguish between transactions
+~~~
+
+To [connect additional SQL clients to the demo cluster]({% link {{ page.version.version }}/cockroach-demo.md %}#connect-an-additional-sql-client-to-the-demo-cluster), in the SQL shell use `\demo ls` to list the connection parameters in the demo cluster:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+\demo ls
+~~~
+
+The output will be similar to:
+
+~~~
+demo@127.0.0.1:26257/defaultdb> \demo ls
+node 1:
+   (webui)    http://127.0.0.1:8080/demologin?password=demo28806&username=demo
+   (cli)      cockroach sql --certs-dir=/Users/myuser/.cockroach-demo -u demo -d defaultdb
+~~~
+
+### Terminal 2
+
+In a second terminal, open another SQL shell to the demo cluster using the `cli` command from the `\demo ls` output:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach sql --certs-dir=/Users/myuser/.cockroach-demo -u demo -d defaultdb
+~~~
+
+In the second SQL shell, set `application_name`:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 2';  -- to distinguish between transactions
+~~~
+
+### Terminal 3
+
+In a third terminal, open another SQL shell to the demo cluster using the `cli` command from the `\demo ls` output:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach sql --certs-dir=/Users/myuser/.cockroach-demo -u demo -d defaultdb
+~~~
+
+In the third SQL shell, set `application_name`:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 3';  -- to distinguish between transactions
+~~~
+
+### DB Console
+
+In a web browser, open the DB Console to the demo cluster using the `webui` address from the `\demo ls` output:
+
+~~~
+http://127.0.0.1:8080/demologin?password=demo28806&username=demo
+~~~
+
+## Understand lock contention
```

Comment:
It seems like each H2 section is a step that should be run sequentially as a procedural (The next step is dependent on having completed the last). You could consider Having ## Step 1. Understand lock contention, ## Step 2. Identify waiting and blocking transactions... and so on

Reply:
Added "Step N" to H2 sections.